packages/material-ui-lab/src/useAutocomplete/useAutocomplete.d.ts

```diff
@@ -18,7 +18,22 @@ export function createFilterOptions<T>(
   config?: CreateFilterOptionsConfig<T>
 ): (options: T[], state: FilterOptionsState<T>) => T[];
 
-export interface UseAutocompleteCommonProps<T> {
+export type AutocompleteFreeSoloValueMapping<T> = T extends true ? string : never;
+
+export type TValue<T, TMultiple, TDisableClearable, TFreeSolo> = TMultiple extends undefined | false
+  ? TDisableClearable extends true
+    ? NonNullable<T | AutocompleteFreeSoloValueMapping<TFreeSolo>>
+    : T | null | AutocompleteFreeSoloValueMapping<TFreeSolo>
+  : Array<T | AutocompleteFreeSoloValueMapping<TFreeSolo>>;
+
+export type TBooleanConstraint = boolean | undefined;
+
+export interface UseAutocompleteProps<
+  T,
+  TMultiple extends TBooleanConstraint,
```

Comment:
Would 
```suggestion
  TMultiple extends boolean | undefined = undefined,
```
work as well?